### PR TITLE
[TVMScript] IRBuilder methods for `For`

### PR DESCRIPTION
**Pull request by cyx-6**
This PR introduces remaining IRBuilder methods for `For`.

Co-authored-by: yongwww <yongcale@gmail.com>

### CODE REVIEW

File: include/tvm/script/ir_builder/tir/frame.h [hunk 1]
```diff
@@ -187,6 +187,51 @@ class BlockFrame : public TIRFrame {
   TVM_DEFINE_MUTABLE_NOTNULLABLE_OBJECT_REF_METHODS(BlockFrame, TIRFrame, BlockFrameNode);
 };
 
+/*!
+ * \brief A frame that represents the for loop.
+ *
+ * \sa BlockInitFrame
```
Comment (junrushao) [inline include/tvm/script/ir_builder/tir/frame.h:193]:
```suggestion
 * \sa ForFrame
```

File: include/tvm/script/ir_builder/tir/frame.h [hunk 2]
```diff
@@ -187,6 +187,51 @@ class BlockFrame : public TIRFrame {
...
+ */
+class ForFrameNode : public TIRFrameNode {
+ public:
+  /*! \brief The for loop generating function type. */
+  using FMakeForLoop =
+      runtime::TypedPackedFunc<tvm::tir::Stmt(Array<tvm::tir::Var>, Array<Range>, tvm::tir::Stmt)>;
```
Comment (junrushao) [inline include/tvm/script/ir_builder/tir/frame.h:199]:
```suggestion
  /*!
   * \brief Functions that generate loop nests.
   * \param loop_vars The loop variables, from outer to inner
   * \param loop_extents The loop extents that correspond to loop variables
   * \param loop_body The loop body
   * \return A stmt, the loop nest
   */
  using FMakeForLoop =
      runtime::TypedPackedFunc<tvm::tir::Stmt(Array<tvm::tir::Var> loop_vars, Array<Range> loop_extents, tvm::tir::Stmt loop_body)>;
```

File: python/tvm/script/ir_builder/tir/frame.py [hunk 3]
```diff
@@ -34,3 +36,10 @@ class PrimFuncFrame(TIRFrame):
 @_register_object("script.ir_builder.tir.BlockFrame")
 class BlockFrame(TIRFrame):
     ...
+
+
+@_register_object("script.ir_builder.tir.ForFrame")
+class ForFrame(TIRFrame):
+    def __enter__(self) -> List[Var]:
```
Comment (junrushao) [inline python/tvm/script/ir_builder/tir/frame.py:43]:
```suggestion
    def __enter__(self) -> Union[Var, List[Var]]:
```

**Review by junrushao — APPROVED** [review]
Otherwise looks pretty good! Let's address my comments and get it in!

File: include/tvm/script/ir_builder/tir/frame.h (hunk 1 above)
Comment (cyx-6) [reply include/tvm/script/ir_builder/tir/frame.h]: updated

File: include/tvm/script/ir_builder/tir/frame.h (hunk 2 above)
Comment (cyx-6) [reply include/tvm/script/ir_builder/tir/frame.h]: updated

File: python/tvm/script/ir_builder/tir/frame.py (hunk 3 above)
Comment (cyx-6) [reply python/tvm/script/ir_builder/tir/frame.py]: updated

**Pull request merged**